osquery/tables/events/darwin/file_changes.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <vector>
+#include <string>
+
+#include <osquery/core.h>
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/events/darwin/fsevents.h"
+
+namespace osquery {
+namespace tables {
+
+/**
+ * @brief Track time, action changes to /etc/passwd
+ *
+ * This is mostly an example EventSubscriber implementation.
+ */
+class FileChangesEventSubscriber
+    : public EventSubscriber<FSEventsEventPublisher> {
+  DECLARE_SUBSCRIBER("file_changes");
+
+ public:
+  void init();
+
+  /**
+   * @brief This exports a single Callback for INotifyEventPublisher events.
+   *
+   * @param ec The EventCallback type receives an EventContextRef substruct
+   * for the INotifyEventPublisher declared in this EventSubscriber subclass.
+   *
+   * @return Was the callback successful.
+   */
+  Status Callback(const FSEventsEventContextRef& ec, const void* extra_data);
+};
+
+/**
+ * @brief Each EventSubscriber must register itself so the init method is
+ *called.
+ *
+ * This registers FileChangesEventSubscriber into the osquery EventSubscriber
+ * pseudo-plugin registry.
+ */
+REGISTER(FileChangesEventSubscriber, "event_subscriber", "file_changes");
+
+void FileChangesEventSubscriber::init() {
+  auto& c = Config::getInstance();
+  const auto& file_map = c.getThreatFiles();
```

Comment:
```cpp
auto file_map =  Config::getInstance().getThreadFiles();
```

Also, https://github.com/facebook/osquery/blob/master/include/osquery/config.h#L109 is NOT returning a reference. Is it possible to change this method to (1) return a const reference to the threat data... maybe that doesn't make sense, but (2) be static so you can write `Config::getThreadFiles()`